src/client/components/Autocomplete/Autocomplete.tsx

```diff
@@ -1,54 +1,103 @@
+  onSave: (value: string | any) => void
 }
 
 const Autocomplete: React.FC<Props> = (props: Props) => {
-  const { value, disabled, onChange, listbox, name, maxItems } = props
+  const { value, items, disabled, name, onInputValueChange, labelKey, onSave } = props
+
+  const [selectedItem, setSelectedItem] = useState(null)
+  const [inputValue, setInputValue] = useState(value)
 
-  if (disabled) return <div className="text-input__readonly-view ">{value}</div>
+  useEffect(() => {
+    setInputValue(value)
+  }, [value])
 
-  const defaultStyles = {
-    item: 'autocomplete__item',
-    listbox: 'autocomplete__listbox',
-    input: 'text-input__input-field',
+  // Handle saving on input field blur or item selected
+  const _onStateChange: (changes: UseComboboxStateChange<any>) => void = (changes) => {
+    if (
+      [UseComboboxStateChangeTypes.InputBlur, UseComboboxStateChangeTypes.ItemClick].includes(
+        changes.type as UseComboboxStateChangeTypes
+      )
+    ) {
+      onSave(changes.selectedItem ?? inputValue)
+    }
   }
 
+  const { isOpen, getMenuProps, getInputProps, getComboboxProps, highlightedIndex, getItemProps } = useCombobox({
+    onInputValueChange(changes) {
+      setInputValue(changes.inputValue)
+      onInputValueChange(changes)
+    },
+    items,
+    itemToString(item) {
+      return item?.[labelKey] ?? item ?? ''
+    },
+    selectedItem,
+    onSelectedItemChange: ({ inputValue, selectedItem: newSelectedItem }) => {
+      setInputValue(inputValue)
+      setSelectedItem(newSelectedItem)
+    },
+    onStateChange: _onStateChange,
+  })
+
   return (
-    <div className={name}>
-      <Turnstone
-        maxItems={maxItems}
-        onChange={onChange}
-        disabled={disabled}
-        text={value}
-        onSelect={onChange}
-        styles={{
-          ...defaultStyles,
-        }}
-        id={name}
-        listbox={listbox}
-        typeahead={false}
+    <div {...getComboboxProps()} className={classNames('autocomplete', { [name]: name })}>
+      <TextInput
+        {...getInputProps({
+          value: inputValue,
+          disabled,
+          className: 'text-input__input-field',
+        })}
       />
+      <div className="autocomplete-dropdown" {...getMenuProps()}>
+        {isOpen &&
+          items.map((item, index) => {
+            return (
+              <div
+                className={classNames('autocomplete-item', {
+                  highlighted: highlightedIndex === index,
+                  selected: Objects.isEqual(selectedItem, item),
+                })}
+                // eslint-disable-next-line react/no-array-index-key
+                key={`${item[labelKey] ?? item}${index}`}
+                {...getItemProps({ item, index })}
+              >
+                {item[labelKey] ?? item}
+              </div>
+            )
+          })}
+      </div>
     </div>
   )
 }
 
 Autocomplete.defaultProps = {
-  name: 'autocomplete',
-  maxItems: 15,
+  name: undefined,
```

Comment:
if it's optional, is this default undefined value needed?

Reply:
we get error otherwise that it's missing defaultProps definition

Reply:
ok!